Security

denma98 opened the pull request:
This solves issue [#468](https://github.com/tattle-made/feluda/issues/468#issue-2765488085)
## Changes Made
1. **Added pip-audit checks** for all Python projects:
   - Finds all `pyproject.toml` files in the repository
   - Converts them to `requirements.txt` using `uv pip compile.`
   - Runs `pip-audit` on the generated requirements files
   - Fails the check if any vulnerabilities are found

2. **Added Deptry dependency checks**:
   - Scans for unused or missing dependencies in Python projects
   - Provides detailed output about dependency issues
   - Fails the check if problems are found (it finds all the issues but still passes the test)

aatmanvaidya commented:
hi @denma98 can you walk me through some `deptry` changes?
as what all have you defined in the `pyproject.toml` file?

denma98 commented:
Mainly, deptry was detecting our custom packages feluda and all the operators. So i added them to our custom packages so that deptry ignores them. Also i added ignore for notebook and /src folder.

denma98 commented:
Also let me check why this is failing, it was working fine on my end.
